src/tests/unit/components/Envelope.vue.spec.js

```diff
@@ -0,0 +1,321 @@
+ */
+
+import {createLocalVue, shallowMount} from '@vue/test-utils'
+
+import Nextcloud from '../../../mixins/Nextcloud'
+import Envelope from '../../../components/Envelope'
+import Vuex from 'vuex';
+
+const localVue = createLocalVue()
+const $route = {
+	params: {
+		id: 1
+	}
+}
+
+localVue.use(Vuex)
+localVue.mixin(Nextcloud)
+
+describe('Envelope', () => {
+	let actions
+	let getters
+	let store
+
+	beforeEach(() => {
+		actions = {}
+		getters = {
+			accounts: () => [
+				{
+					id: 123,
+				},
+			],
+			getAccount: () => (id) => ({}),
+			getEnvelopeTags: () => (id) => ([]),
+		}
+		store = new Vuex.Store({
+			actions,
+			getters,
+		})
+	})
+	it('allows toggling seen flag without ACLs', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: undefined,
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasSeenAcl).toBe(true)
+	})
+
+	it('disallows toggling seen flag without s ACL right', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 'x',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasSeenAcl).toBe(false)
+	})
+
+	it('allows toggling seen flag with s ACL right', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 's',
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasSeenAcl).toBe(true)
+	})
+	it('allows toggling archive action without ACLs', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: undefined,
+					accountId: 123,
+					from: [{email:'info@test.com'}],
+					flags: { seen:false, flagged:false, $junk:false, answered:false, hasAttachments:false, draft:false, },
+				},
+			},
+			store,
+			localVue,
+		})
+
+		expect(view.vm.hasArchiveAcl).toBe(true)
+	})
+	it('source mailbox has t ACLs for archiving', () => {
+		const view = shallowMount(Envelope, {
+			mocks: {
+				$route,
+			},
+			propsData: {
+				account: {},
+				mailbox: {
+					specialRole:'',
+				},
+				data: {
+					myAcls: 't',
```

Comment:
```suggestion
					myAcls: 'te',
```

You also need to retrieve the ACLs for the target mailbox from the store and make them contain `i` for this test. You could make use of the field `this.account.archiveMailboxId` of `Envelope.vue`.

Reply:
i delivered everything, but this one, i dont know how to test this one

Reply:
Done